### PR TITLE
Updated `Bleed` to accept responsive spacing

### LINKED ISSUE #7792: [Layout foundations] Add responsive spacing to `Bleed`

### PR DESCRIPTION
**Pull request by aveline**
### WHY are these changes introduced?

Resolves #7792 and unblocks #7354 

### WHAT is this pull request doing?
- Add support for responsive spacing values to be passed in to `Bleed`
- Created the responsive custom properties manually instead of using the mixin because the mixin doesn't currently support applying the negative values with `calc()`. Not sure it's a use case we will run into enough to warrant adding the complexity to the mixin to support this. However if that is something we find ourselves needing to do repeatedly then worth revisiting the mixin.

### CODE REVIEW

File: polaris-react/src/components/Bleed/Bleed.stories.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ export function Default() {
 
 export function WithVerticalDirection() {
   return (
-    <Box background="surface" padding="4">
+    <Box background="surface-success" padding="4">
```
Comment (kyledurand) [inline polaris-react/src/components/Bleed/Bleed.stories.tsx]: Maybe we match these styles to the [styleguide](https://www.figma.com/file/5oEi1wm73KhGLN0VS02oYm/Layout-components?node-id=2765%3A32014&t=WnCODY49P5Sq7HFA-4) examples? I think it's clearer what's happening with just the border vs a bg color

File: polaris-react/src/components/Bleed/Bleed.scss [hunk 2]
```diff
@@ -1,8 +1,56 @@
...
+  --pc-bleed-margin-inline-start-sm: var(--pc-bleed-margin-inline-start-xs);
+  --pc-bleed-margin-inline-start-md: var(--pc-bleed-margin-inline-start-sm);
+  --pc-bleed-margin-inline-start-lg: var(--pc-bleed-margin-inline-start-md);
+  --pc-bleed-margin-inline-start-xl: var(--pc-bleed-margin-inline-start-lg);
+  --pc-bleed-margin-inline-end-xs: initial;
+  --pc-bleed-margin-inline-end-sm: var(--pc-bleed-margin-inline-end-xs);
+  --pc-bleed-margin-inline-end-md: var(--pc-bleed-margin-inline-end-sm);
+  --pc-bleed-margin-inline-end-lg: var(--pc-bleed-margin-inline-end-md);
+  --pc-bleed-margin-inline-end-xl: var(--pc-bleed-margin-inline-end-lg);
+  margin-block-start: calc(-1 * var(--pc-bleed-margin-block-start-xs));
```
Comment (kyledurand) [inline polaris-react/src/components/Bleed/Bleed.scss:24]: I like the idea of handling these manually for now as well 👍

**Review by kyledurand — APPROVED** [review]
👍

**Review by laurkim — APPROVED** [review]
Nice!

**Pull request merged**